docs/hugo/content/architecture-decision-records/ADR-2022-02-Backward-Compatibility.md

```diff
@@ -1,17 +1,88 @@
 # Context
 
-As we close in on a beta release, we want to modify the version prefix used for code-generated resources.
+As we close in on a beta release, we are considering modification of the version prefix used for code-generated resources.
 
-We don't want to maroon existing users of Azure Service Operator (ASO) by making this a breaking change.
+## Option 1: Do nothing
 
-Cross version compatibility between ARM versions of a resource (that is, between `v1alpha1api20180101` and `v1alpha1api20200101`) is already implemented and well tested, so we want to leverage that existing infrastructure.
+Leave all generated resources using the alpha prefix that is currently in use.
+
+Pros
+
+* Easy to implement (nothing to do) and quick to deliver
+
+Cons
+
+* Introduces an inconsistency between the status of ASO (beta) and the labelling of the resources (alpha).
+* Might be perceived by some users as indicating these particular resources are still in alpha, and therefore pose significant a barrier to adoption of ASO.
+
+## Option 2: Move all resources to beta
+
+Change the generated prefix for all resources
+
+Pros
+
+* Easy to implement (change the prefix, update affected tests, and regenerate all files), and quick to deliver
+
+Cons
+
+* Existing clusters with custom resources (CRs) reliant on ASO would break because ASO would no longer support the older version.  
+  Mitigation would require deleting the existing resource and creating a new CR with an updated version for the resource.
```

Comment:
```suggestion
  Migration would require deleting the existing resource and creating a new CR with an updated version for the resource.
```

Reply:
Not sure if that was meant to read "Migration"? I guess "Mitigation" also works :)